### DISCUSSION TITLE
make_trade() is needing to Approve same crypto more than once

### DISCUSSION THREAD

**Discussion by sqhell** [uniswap-python/uniswap-python#140] [Q&A]
I am using make_trade() and even though 7 days ago a crypto transaction for Approve happened, I needed to do it again today. Is there something wrong, or something I need to do to check if crypto has already been approved?

**Comment by liquid-8** [comment]
Actually, manual approves aren't obligatory at all. `make_trade()` auto-approves `input_token` if it wasn't approved before. Feel free to post your code and/or txns.
